src/pages/permissions/PermissionIdpGroups.tsx

```diff
@@ -0,0 +1,272 @@
+import SelectedTableNotification from "components/SelectedTableNotification";
+import { FC, useEffect, useState } from "react";
+import { queryKeys } from "util/queryKeys";
+import useSortTableData from "util/useSortTableData";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import NotificationRow from "components/NotificationRow";
+import HelpLink from "components/HelpLink";
+import { useDocs } from "context/useDocs";
+import PermissionGroupsFilter from "./PermissionGroupsFilter";
+import { pluralize } from "util/instanceBulkActions";
+import { fetchIdpGroups } from "api/auth-idp-groups";
+import CreateIdpGroupPanel from "./panels/CreateIdpGroupPanel";
+import IdpGroupActions from "./actions/IdpGroupActions";
+import BulkDeleteIdpGroupsBtn from "./actions/BulkDeleteIdpGroupsBtn";
+import EditIdpGroupPanel from "./panels/EditIdpGroupPanel";
+
+const PermissionIdpGroups: FC = () => {
+  const notify = useNotify();
+  const {
+    data: groups = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.idpGroups],
+    queryFn: fetchIdpGroups,
+  });
+  const docBaseLink = useDocs();
+  const panelParams = usePanelParams();
+  const [search, setSearch] = useState("");
+  const [selectedGroupNames, setSelectedGroupNames] = useState<string[]>([]);
+
+  if (error) {
+    notify.failure("Loading provider groups failed", error);
+  }
+
+  useEffect(() => {
+    if (panelParams.idpGroup) {
+      setSelectedGroupNames([panelParams.idpGroup]);
+    }
+  }, [panelParams.idpGroup]);
+
+  const headers = [
+    { content: "Name", className: "name", sortKey: "name" },
+    {
+      content: "Groups",
+      sortKey: "groups",
+      className: "u-align--right",
+    },
+    { "aria-label": "Actions", className: "u-align--right actions" },
+  ];
+
+  const filteredGroups = groups.filter(
+    (idpGroup) => !search || idpGroup.name.toLowerCase().includes(search),
+  );
+
+  const selectedGroups = groups.filter((group) =>
+    selectedGroupNames.includes(group.name),
+  );
+
+  const groupsForRows =
+    panelParams.panel && panelParams.panel !== panels.createIdpGroup
+      ? selectedGroups
+      : filteredGroups;
+
+  const rows = groupsForRows.map((idpGroup) => {
+    return {
+      name: idpGroup.name,
+      className: "u-row",
+      columns: [
+        {
+          content: idpGroup.name,
+          role: "cell",
+          "aria-label": "Name",
+          className: "u-truncate",
+          title: idpGroup.name,
+        },
+        {
+          content: idpGroup.groups.length,
+          role: "cell",
+          className: "u-align--right",
+          "aria-label": "Number of mapped groups",
+        },
+        {
+          className: "actions u-align--right",
+          content: <IdpGroupActions idpGroup={idpGroup} />,
+          role: "cell",
+          "aria-label": "Actions",
```

Comment:
I think we can show the icons directly, no need for a context menu. @piperdeck

![Screenshot from 2024-04-26 12-03-49](https://github.com/canonical/lxd-ui/assets/1155472/ad4a71ba-db13-43f0-9294-d2595f0063cb)